test-util/src/main/scala/io/buoyant/test/Exceptions.scala

```diff
@@ -0,0 +1,10 @@
+package io.buoyant.test
+
+import org.scalatest.Assertions
+
+trait Exceptions extends Assertions {
+
+  def assertThrows[T <: AnyRef: Manifest](f: => Any): Unit = {
```

Comment:
might be worth adding a comment here explaining that we provide this because `intercept` triggers value-discarded warnings?